### The relation between Pclass and Cabin

Here we will check if there is a relationship between Pclass and Cabin the hypothesis i had is if it is a third class there is no Cabin value 

In [ ]:
data.groupby('Pclass')['Cabin'].apply(lambda x: x.isnull().mean())

Pclass
1    0.185185
2    0.913043
3    0.975560
Name: Cabin, dtype: float64

### The relation between Pclass and Embarked

Here we will check if there is a relationship between Pclass and Embarked.

**Observation**

- Class 1: 18.5% missing
- Class 2: 91.3% missing
- Class 3: 97.6% missing
- class 1 mostly has Cabin recorded, and classes 2 and 3 are both overwhelmingly missing, at nearly the same rate.

In [ ]:
data.groupby('Embarked')['Pclass'].value_counts(normalize=True)

Embarked  Pclass
C         1         0.505952
          3         0.392857
          2         0.101190
Q         3         0.935065
          2         0.038961
          1         0.025974
S         3         0.548137
          2         0.254658
          1         0.197205
Name: proportion, dtype: float64